taf/models/types.py

```diff
@@ -25,12 +25,22 @@ def value(self) -> str:
         return self.tag if self.tag else self.hash
 
     @classmethod
-    def from_hash(cls, hash: Optional[str]):
-        return cls(hash) if hash is not None else None  # type: ignore
+    def from_hash(cls, hash: Optional[Commitish | str]):
+        """
+        Initialize cls from `hash`.
+        If `hash` is already `Commitish`, returns self
+        """
+        if hash is None:
+            return None
+        if isinstance(hash, str):
+            return cls(hash)  # type: ignore
+        return hash  # type: ignore
 
     def __eq__(self, other):
         if other is None:
             return False
+        if isinstance(other, str):
```

Comment:
I tried to avoid adding this and convert all string SHAs to Commitish objects. I think it's fine to support this to avoid issues in code that interacts with TAF.